packages/nimble-components/src/menu-button/tests/menu-button.spec.ts

```diff
@@ -256,86 +227,92 @@ describe('MenuButton', () => {
-                toggleListener.spy
+                'beforetoggle',
+                spy
             );
+            const togglePromise = waitForEvent(element, 'toggle', spy);
+
+            expect(element.open).toEqual(false);
+            expect(spy).not.toHaveBeenCalled();
+
+            pageObject.clickMenuButton();
+            await Promise.all([togglePromise, beforeTogglePromise]);
+
             const expectedDetails: MenuButtonToggleEventDetail = {
                 newState: true,
                 oldState: false
             };
-
-            pageObject.clickMenuButton();
-            await beforeToggleListener.promise;
-            expect(beforeToggleListener.spy).toHaveBeenCalledTimes(1);
-            const event = beforeToggleListener.spy.calls.first()
-                .args[0] as CustomEvent;
-            expect(event.detail).toEqual(expectedDetails);
-            beforeToggleListener.spy.calls.reset();
-
-            await toggleListener.promise;
-            expect(beforeToggleListener.spy).not.toHaveBeenCalled();
-            expect(toggleListener.spy).toHaveBeenCalledTimes(1);
+            expect(spy).toHaveBeenCalledTimes(2);
+            const beforetoggleEvent = spy.calls.argsFor(0)[0];
+            expect(beforetoggleEvent.type).toEqual('beforetoggle');
+            expect(beforetoggleEvent.detail).toEqual(expectedDetails);
+            const toggleEvent = spy.calls.argsFor(1)[0];
+            expect(toggleEvent.type).toEqual('toggle');
+            expect(toggleEvent.detail).toEqual(expectedDetails);
         });
 
         it("should fire 'beforetoggle' event before the menu is closed", async () => {
             element.open = true;
             await connect();
-            const toggleListener = createEventListener(element, 'toggle');
-            const beforeToggleListener = createBeforeToggleListener(
+            const spy = jasmine.createSpy<MenuButtonToggleEventHandler>();
+            const beforeTogglePromise = waitForEvent(
                 element,
-                true,
-                toggleListener.spy
+                'beforetoggle',
+                spy
             );
+            const togglePromise = waitForEvent(element, 'toggle', spy);
+
+            expect(element.open).toEqual(true);
+            expect(spy).not.toHaveBeenCalled();
+
+            pageObject.clickMenuButton();
+            await Promise.all([togglePromise, beforeTogglePromise]);
+
             const expectedDetails: MenuButtonToggleEventDetail = {
                 newState: false,
                 oldState: true
             };
-
-            pageObject.clickMenuButton();
-            await beforeToggleListener.promise;
-            expect(beforeToggleListener.spy).toHaveBeenCalledTimes(1);
-            const event = beforeToggleListener.spy.calls.first()
-                .args[0] as CustomEvent;
-            expect(event.detail).toEqual(expectedDetails);
-            beforeToggleListener.spy.calls.reset();
```

Comment:
This isn't a useful way to track the spy as createEventListener only fired the event once. the spy call count wouldn't have ever gone up

Reply:
Was on the fence about renaming `waitForEvent` to something like `waitForSingleEvent`.
I'm thinking `createEventListener` is more misleading to believing it would potentially trigger multiple times / continuously compared to `waitForEvent` so didn't make that change.

Reply:
The name `waitForEvent` seems good enough to me. It returns a Promise, not an Observable or callback or something where there would be an expectation that it'd fire multiple times.